node/tests/p2p_test.go

```diff
@@ -188,42 +188,51 @@ func TestRestartNodeDiscovery(t *testing.T) {
 		nodes[index] = sw.NewFullNode(nodesConfig...)
 	}
 
-	sub, err := nodes[0].Host.EventBus().Subscribe(&event.EvtPeerConnectednessChanged{})
+	identitySub, err := nodes[0].Host.EventBus().Subscribe(&event.EvtPeerIdentificationCompleted{})
 	require.NoError(t, err)
-	defer sub.Close()
+	defer identitySub.Close()
+
 	for index := 0; index < fullNodes; index++ {
 		require.NoError(t, nodes[index].Start(ctx))
-		assert.Equal(t, *addr, nodes[index].Bootstrappers[0])
 		assert.True(t, nodes[index].Host.Network().Connectedness(addr.ID) == network.Connected)
 	}
 
 	// wait until full nodes connect each other
-	e := <-sub.Out()
-	connStatus := e.(event.EvtPeerConnectednessChanged)
+	e := <-identitySub.Out()
+	connStatus := e.(event.EvtPeerIdentificationCompleted)
 	id := connStatus.Peer
 	if id != nodes[1].Host.ID() {
 		t.Fatal("unexpected peer connected")
 	}
-	assert.True(t, nodes[0].Host.Network().Connectedness(id) == network.Connected)
+	require.True(t, nodes[0].Host.Network().Connectedness(id) == network.Connected)
 
 	// create one more node with disabled discovery
 	nodesConfig[1] = node.WithPeersLimit(0)
 	node := sw.NewFullNode(nodesConfig...)
+	connectSub, err := nodes[0].Host.EventBus().Subscribe(&event.EvtPeerConnectednessChanged{})
+	require.NoError(t, err)
+	defer connectSub.Close()
+
+	// disconnect and unlink peers in order to receive disconnected event
+	// NOTE: Consistence is very important here. We have to unlink peers and only after that call disconnect
+	// otherwise pubsub could re-establish the connection.
```

Comment:
```suggestion
	// NOTE: Order is very important here. We have to unlink peers, and only after that call disconnect,
	// otherwise pubsub could re-establish the connection.
```